src/net.h

```diff
@@ -614,12 +614,27 @@ class CNetMessage {
     int readData(const char *pch, unsigned int nBytes);
 };
 
+/** The TransportSerializer prepares messages for the network transport
+ */
+class TransportSerializer {
+public:
+    // prepare message for transport (insert header, etc.)
```

Comment:
nit: Comments could be better for public interfaces, parent class could list the abstract operations a transport protocol may realize like header construction, error-correction computation, payload encryption and then for transport subclass would list the concrete sequence of operations implemented?

Reply:
Slightly improved...